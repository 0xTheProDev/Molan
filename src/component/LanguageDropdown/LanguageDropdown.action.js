// @flow
import React from 'react';
import { LanguageDropdownOption, onLanguageChangeCallbackParam } from './LanguageDropdown.flow';

export function onLanguageChangeCallback(onLanguageChange: onLanguageChangeCallbackParam) {
  return function onChangeHandler(event: React.SyntheticEvent, data: LanguageDropdownOption) {
    onLanguageChange(data.value);
  }
}
