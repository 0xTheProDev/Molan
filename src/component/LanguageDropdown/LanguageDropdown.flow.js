// @flow
import React from 'react';

export type onLanguageChangeCallbackParam = (language: string) => void;

export type LanguageDropdownProps = {
  DefaultLanguage: string,
  onLanguageChange: onLanguageChangeCallbackParam,
};

export type LanguageDropdownOption = {
  key: string,
  text: string,
  value: string,
};

export type LanguageDropdownInternalProps = {
  LanguageDropdownPlaceholder: string,
  DefaultLanguage: string,
  LanguageOptions: Array<LanguageDropdownOption>,
  onLanguageChange: (event: React.SyntheticEvent, selectedLanguage: LanguageDropdownOption) => void,
};
