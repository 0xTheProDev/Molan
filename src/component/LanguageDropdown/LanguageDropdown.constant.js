// @flow
import { LANGUAGES, COMPILER_OPTIONS } from 'constants/templates';

import { LanguageDropdownOption } from './LanguageDropdown.flow';

export const LANGUAGE_DROPDOWN_OPTIONS: Array<LanguageDropdownOption> =  LANGUAGES.map(language => {
  return {
    key: language,
    value: language,
    text: COMPILER_OPTIONS[language],
  };
});
