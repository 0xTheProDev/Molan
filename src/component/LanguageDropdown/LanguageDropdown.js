// @flow
import React from 'react';
import { Select } from 'semantic-ui-react';
import { noop } from 'util/common';

import { LanguageDropdownInternalProps } from './LanguageDropdown.flow';
import { LANGUAGE_DROPDOWN_OPTIONS } from './LanguageDropdown.constant';

export function LanguageDropdown(props: LanguageDropdownInternalProps) {
  const {
    LanguageDropdownPlaceholder,
    LanguageOptions,
    DefaultLanguage,
    onLanguageChange,
  } = props;

  return (
    <Select
      placeholder={LanguageDropdownPlaceholder}
      options={LanguageOptions}
      value={DefaultLanguage}
      onChange={onLanguageChange}
    />
  );
}

LanguageDropdown.defaultProps = {
  LanguageDropdownPlaceholder: 'Select a language',
  LanguageOptions: LANGUAGE_DROPDOWN_OPTIONS,
  onLanguageChange: noop,
};
