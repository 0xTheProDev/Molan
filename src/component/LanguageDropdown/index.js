// @flow
import React from 'react';

import { LanguageDropdownProps } from './LanguageDropdown.flow';
import { onLanguageChangeCallback } from './LanguageDropdown.action';
import { LanguageDropdown } from './LanguageDropdown';

export default function LanguageDropdownFactory({ DefaultLanguage, onLanguageChange }: LanguageDropdownProps) {
  const onChangeHandler = onLanguageChangeCallback(onLanguageChange);

  return (
    <LanguageDropdown
      DefaultLanguage={DefaultLanguage}
      onLanguageChange={onChangeHandler}
    />
  );
}

export { LanguageDropdownProps, LanguageDropdownFactory as LanguageDropdown };
