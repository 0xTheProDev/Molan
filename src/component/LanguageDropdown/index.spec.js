import React from 'react';
import { DropdownItem } from 'semantic-ui-react';

import LanguageDropdown from './index';
import { LANGUAGE_DROPDOWN_OPTIONS } from './LanguageDropdown.constant';

describe('Component::LanguageDropdown::Integration Suite', () => {
  let
    onLanguageChangeStub,
    minProps;

  beforeAll(() => {
    onLanguageChangeStub = jest.fn();

    minProps = {
      DefaultLanguage: LANGUAGE_DROPDOWN_OPTIONS[0].value,
      onLanguageChange: onLanguageChangeStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = mount(<LanguageDropdown {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onLanguageChange with new language when selected one', () => {
    const wrapper = mount(<LanguageDropdown {...minProps} />);

    const lastDropdownItem = wrapper
      .find(DropdownItem)
      .last();

    lastDropdownItem.simulate('click');

    expect(onLanguageChangeStub).toHaveBeenCalledWith(lastDropdownItem.props().value);
  });
});
