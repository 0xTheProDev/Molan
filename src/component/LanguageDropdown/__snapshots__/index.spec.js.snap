// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::LanguageDropdown::Integration Suite should render with base structure 1`] = `
<LanguageDropdownFactory
  DefaultLanguage="c"
  onLanguageChange={[MockFunction]}
>
  <LanguageDropdown
    DefaultLanguage="c"
    LanguageDropdownPlaceholder="Select a language"
    LanguageOptions={
      Array [
        Object {
          "key": "c",
          "text": "C (GCC 6.3)",
          "value": "c",
        },
        Object {
          "key": "cpp",
          "text": "C++ (G++ 6.3)",
          "value": "cpp",
        },
        Object {
          "key": "java",
          "text": "Java (OpenJDK 0.8)",
          "value": "java",
        },
        Object {
          "key": "python",
          "text": "Python (Python 2.7)",
          "value": "python",
        },
        Object {
          "key": "python3",
          "text": "Python (Python 3.6)",
          "value": "python3",
        },
        Object {
          "key": "mips",
          "text": "MIPS-IV (Simulator 0.4)",
          "value": "mips",
        },
        Object {
          "key": "verilog",
          "text": "Verilog (iVerilog 10.0)",
          "value": "verilog",
        },
      ]
    }
    onLanguageChange={[Function]}
  >
    <Select
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "c",
            "text": "C (GCC 6.3)",
            "value": "c",
          },
          Object {
            "key": "cpp",
            "text": "C++ (G++ 6.3)",
            "value": "cpp",
          },
          Object {
            "key": "java",
            "text": "Java (OpenJDK 0.8)",
            "value": "java",
          },
          Object {
            "key": "python",
            "text": "Python (Python 2.7)",
            "value": "python",
          },
          Object {
            "key": "python3",
            "text": "Python (Python 3.6)",
            "value": "python3",
          },
          Object {
            "key": "mips",
            "text": "MIPS-IV (Simulator 0.4)",
            "value": "mips",
          },
          Object {
            "key": "verilog",
            "text": "Verilog (iVerilog 10.0)",
            "value": "verilog",
          },
        ]
      }
      placeholder="Select a language"
      value="c"
    >
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        closeOnEscape={true}
        deburr={false}
        icon="dropdown"
        minCharacters={1}
        noResultsMessage="No results found."
        onChange={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "key": "c",
              "text": "C (GCC 6.3)",
              "value": "c",
            },
            Object {
              "key": "cpp",
              "text": "C++ (G++ 6.3)",
              "value": "cpp",
            },
            Object {
              "key": "java",
              "text": "Java (OpenJDK 0.8)",
              "value": "java",
            },
            Object {
              "key": "python",
              "text": "Python (Python 2.7)",
              "value": "python",
            },
            Object {
              "key": "python3",
              "text": "Python (Python 3.6)",
              "value": "python3",
            },
            Object {
              "key": "mips",
              "text": "MIPS-IV (Simulator 0.4)",
              "value": "mips",
            },
            Object {
              "key": "verilog",
              "text": "Verilog (iVerilog 10.0)",
              "value": "verilog",
            },
          ]
        }
        placeholder="Select a language"
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        selection={true}
        value="c"
        wrapSelection={true}
      >
        <Ref
          innerRef={
            Object {
              "current": <div
                aria-expanded="false"
                class="ui selection dropdown"
                role="listbox"
                tabindex="0"
              >
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  class="text"
                  role="alert"
                >
                  C (GCC 6.3)
                </div>
                <i
                  aria-hidden="true"
                  class="dropdown icon"
                />
                <div
                  class="menu transition"
                >
                  <div
                    aria-checked="true"
                    aria-selected="true"
                    class="active selected item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      C (GCC 6.3)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      C++ (G++ 6.3)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      Java (OpenJDK 0.8)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      Python (Python 2.7)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      Python (Python 3.6)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      MIPS-IV (Simulator 0.4)
                    </span>
                  </div>
                  <div
                    aria-checked="false"
                    aria-selected="false"
                    class="item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      Verilog (iVerilog 10.0)
                    </span>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <div
                  aria-expanded="false"
                  class="ui selection dropdown"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-atomic="true"
                    aria-live="polite"
                    class="text"
                    role="alert"
                  >
                    C (GCC 6.3)
                  </div>
                  <i
                    aria-hidden="true"
                    class="dropdown icon"
                  />
                  <div
                    class="menu transition"
                  >
                    <div
                      aria-checked="true"
                      aria-selected="true"
                      class="active selected item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        C (GCC 6.3)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        C++ (G++ 6.3)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Java (OpenJDK 0.8)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Python (Python 2.7)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Python (Python 3.6)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        MIPS-IV (Simulator 0.4)
                      </span>
                    </div>
                    <div
                      aria-checked="false"
                      aria-selected="false"
                      class="item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        Verilog (iVerilog 10.0)
                      </span>
                    </div>
                  </div>
                </div>,
              }
            }
          >
            <div
              aria-expanded={false}
              className="ui selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="listbox"
              tabIndex={0}
            >
              <div
                aria-atomic={true}
                aria-live="polite"
                className="text"
                role="alert"
              >
                C (GCC 6.3)
              </div>
              <Icon
                as="i"
                className=""
                name="dropdown"
                onClick={[Function]}
              >
                <i
                  aria-hidden="true"
                  className="dropdown icon"
                  onClick={[Function]}
                />
              </Icon>
              <DropdownMenu>
                <div
                  className="menu transition"
                >
                  <DropdownItem
                    active={true}
                    key="c"
                    onClick={[Function]}
                    selected={true}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="C (GCC 6.3)"
                    value="c"
                  >
                    <div
                      aria-checked={true}
                      aria-selected={true}
                      className="active selected item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        C (GCC 6.3)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="cpp"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="C++ (G++ 6.3)"
                    value="cpp"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        C++ (G++ 6.3)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="java"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="Java (OpenJDK 0.8)"
                    value="java"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Java (OpenJDK 0.8)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="python"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="Python (Python 2.7)"
                    value="python"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Python (Python 2.7)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="python3"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="Python (Python 3.6)"
                    value="python3"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Python (Python 3.6)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="mips"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="MIPS-IV (Simulator 0.4)"
                    value="mips"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        MIPS-IV (Simulator 0.4)
                      </span>
                    </div>
                  </DropdownItem>
                  <DropdownItem
                    active={false}
                    key="verilog"
                    onClick={[Function]}
                    selected={false}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="Verilog (iVerilog 10.0)"
                    value="verilog"
                  >
                    <div
                      aria-checked={false}
                      aria-selected={false}
                      className="item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        Verilog (iVerilog 10.0)
                      </span>
                    </div>
                  </DropdownItem>
                </div>
              </DropdownMenu>
            </div>
          </RefFindNode>
        </Ref>
      </Dropdown>
    </Select>
  </LanguageDropdown>
</LanguageDropdownFactory>
`;
